gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java

```diff
@@ -239,31 +241,49 @@ private void handleRequest(final HttpServerRequest req, final Buffer bodyData, f
         final LoggingHandler loggingHandler = new LoggingHandler(loggingResourceManager, logAppenderRepository, req, vertx.eventBus());
 
         final String uniqueId = req.headers().get("x-rp-unique_id");
-        final String timeout = req.headers().get("x-timeout");
+        final String timeoutRaw = req.headers().get("x-timeout");
+        long timeoutMs = (timeoutRaw != null) ? parseLong(timeoutRaw) : rule.getTimeout();
         final long startTime = monitoringHandler.startRequestMetricTracking(rule.getMetricName(), req.uri());
 
+        retryPerformRequest(req, bodyData, targetUri, log, profileHeaderMap, authHeader.orElse(null),
+            afterHandler, loggingHandler, uniqueId, timeoutMs, startTime);
+    }
+
+    private void retryPerformRequest(
+        HttpServerRequest req, Buffer bodyData, String targetUri, Logger log, Map<String, String> profileHeaderMap,
+        AuthHeader authHeader, @Nullable Handler<Void> afterHandler, LoggingHandler loggingHandler,
+        String uniqueId, long timeoutMs, long startTime
+    ){
         client.request(req.method(), port, rule.getHost(), targetUri, new Handler<>() {
             @Override
             public void handle(AsyncResult<HttpClientRequest> event) {
                 req.resume();
 
                 if (event.failed()) {
-                    log.warn("Problem to request {}: {}", targetUri, event.cause());
+                    Throwable ex = event.cause();
+                    long postponeMs = 1000;
+                    long remainingTimeMs = timeoutMs - postponeMs;
+                    if (remainingTimeMs > 0 && ex instanceof ConnectionPoolTooBusyException) {
```

Comment:
What about the sizing of the ConnectionPool itself, must it be tuned accordingly to reduce the probability of this occurence?

Reply:
Feel free to open a PR for it :)